lib/reservations.rb

```diff
@@ -0,0 +1,87 @@
+require 'pry'
+require 'date'
+
+module Hotel
+  class Reservations
+    attr_reader :all_rooms, :all_reservations
+    def initialize
+      @all_rooms = []
+      @all_reservations = []
+      20.times do |i|
+        i += 1
+        @all_rooms << Hotel::Room.new(i)
+      end
+    end
+
+    def make_reservation(check_in,check_out,num_rooms, block_name:false)
+      date_range = DateRange.new(check_in,check_out)
+      id = (@all_reservations.length + 1)
+      rooms_available = check_availability(check_in,check_out)
+      enough_rooms?(rooms_available, num_rooms, block_name)
+      rooms_to_book = rooms_available.shift(num_rooms)
+
+      if block_name == false
+        update_room_state(rooms_to_book,num_rooms,true)
+        booking = Booking.new(id,rooms_to_book,date_range, block_info: false)
+      else
+        update_room_state(rooms_to_book,num_rooms,false)
+        booking = Block.new(id,rooms_to_book,date_range,block_name)
+      end
+      @all_reservations << booking
+      return booking
+    end
+
+    def check_reservations(check_in,check_out)
+      booked_rooms = []
+      range = DateRange.new(check_in,check_out).nights_arr
+      range.each do |date|
+        @all_reservations.each do |booking|
+          if booking.date_range.nights_arr.include?(date)
```

Comment:
Instead of looping through every date in the range, it might be a little cleaner to write a method `DateRange#overlap?(other)`. Then this loop could be simplified to:

```ruby
booked_rooms = []
range = DateRange.new(check_in,check_out)
@all_reservations.each do |booking|
  if booking.date_range.overlap?(range)
    # add rooms to booked_rooms
  end
end
return booked_rooms
```

It's doing the same work, but I would argue this version is more readable, and does a better job of separating responsibilities. It also makes testing easier!